### ISSUE TITLE
Consume ISO Strings as well

### ISSUE THREAD

**Issue by langpavel** [tjmehta/graphql-date#2]
Hi, do you plan to support passing ISO serialized date as well? This means [this test](https://github.com/tjmehta/graphql-date/blob/master/test/graphql-date.test.js#L11..L17) will be successful.

Reason: I'm using REST API which serves me dates serialized as string. I wish to pass result as is to express-graphql endpoint and don't care of format.

It can be also useful to pass Numbers as UNIX timestamps, but this is problematic because of much API varies between milliseconds and seconds.

**Comment by tjmehta** [comment]
`Serialize` converts a date to a str - hence why that test fails. Note that parse works fine: https://github.com/tjmehta/graphql-date/blob/master/test/graphql-date.test.js#L48. I happy to accept PRs that support other serialized date values.

**Comment by langpavel** [comment]
I'm not sure if you understand me.

I wish to forward serialized string formatted as ISO date string. Not Date instance.
It's because I'm only forwarding JSON response from foreign API and I'm receiving serialized form.

I need that `serialize` accepts already serialized input.

I'm only read your code, not try, but it is clear from test that your module doesn't work this way.

I'm using this implementation now: https://gist.github.com/langpavel/b30f3d507a47713b0c6e89016e4e9eb7

Question now is if you want to support this behavior, because it's not clean and it's use case specific :-)

**Comment by tjmehta** [comment]
I understand now. You want to support various date formats (not serialized). Hmm, my first thought is to export the different formats as different types. The more the strict the type the more likely it will catch errors in your code. What do you think?

**Comment by langpavel** [comment]
You are correct. _The more the strict the type the more likely it will catch errors in your code..._ I'm observing with native GraphQL types that they are returning `null` instead of throwing an exception if coercion fails.
On the other hand, your code is clean. I like it but it does not fit to my needs. There may be other users and they are conform with your library as is.

I think exporting more types is bad idea, because _it is exactly one domain type_. Some kind of flag or singleton factory should be better..? I don't know ATM.

I suggest let this issue open and **I suggest to others, act** if you need this behavior. There is :+1: voting :-)

You have 116 downloads per month, not much but not bad as well :-)

Thank you for your quick responses. I'm too early with my use case as well so I have no clear idea on this..

**Comment by tjmehta** [comment]
Cool, will do. Thanks for bringing this up

**Comment by langpavel** [comment]
Good to know, I will watch this issue, let me know then. :smile:
